### PR TITLE
Make context optional for evaluation

### PR DESCRIPTION
**Pull request by barney-s**
### Change description

This is to allow simpler compositions and especially compositions that don't use any cloud resources.

### CODE REVIEW

**Comment by xiaoweim** [comment]
Looks good, does this mean that not all composition uses context? (since so far my understanding is that all compositions use context and will generate a plan)

File: experiments/compositions/composition/internal/controller/expander_reconciler.go [hunk 1]
```diff
@@ -424,17 +424,20 @@ func (r *ExpanderReconciler) evaluateAndSavePlan(ctx context.Context, logger log
 	}
 
 	// read context in cr.namespace
+	var contextBytes []byte
 	contextcr := unstructured.Unstructured{}
 	contextcr.SetGroupVersionKind(contextGVK)
 	contextNN := types.NamespacedName{Namespace: cr.GetNamespace(), Name: "context"}
 	if err := r.Get(ctx, contextNN, &contextcr); err != nil {
 		logger.Error(err, "unable to fetch Context CR", "context", contextNN)
```
Comment (justinsb) [inline experiments/compositions/composition/internal/controller/expander_reconciler.go:432]: Can we ignore _only_ `apierrors.IsNotFound(err)`, and still return an error if it's a different error.  The reason it matters is that you don't want e.g. timeout talking to kube-apiserver to cause a different, temporary reconciliation result.

Comment (barney-s) [reply experiments/compositions/composition/internal/controller/expander_reconciler.go:432]: Ah yes. That was the intent.

**Comment by barney-s** [comment]
> Looks good, does this mean that not all composition uses context? (since so far my understanding is that all compositions use context and will generate a plan)

Yes for non-cloud resources, `context` is optional.

**Pull request merged**